libs/hdf-converters/src/converters-from-hdf/caat/reverse-caat-mapper.ts

```diff
@@ -0,0 +1,257 @@
+      return {
+        data: contextualizedEvaluation,
+        filename:
+          datum.filename ??
+          contextualizedEvaluation.data.profiles.at(0)?.name ??
+          'ExecJSON',
+        controls: datum.controls
+      };
+    });
+  }
+
+  // ensure we're using Windows style newlines and fit within the maximum length
+  fix(str: string | null | undefined): string {
+    return (str ?? '')
+      .replace(/(\r\n|\n|\r)/gmu, '\r\n')
+      .slice(0, FromHDFToCAATMapper.MaxCellSize);
+  }
+
+  createCaveat(hdf: HDFControl): string {
+    const caveat = hdf.descriptions.caveat
+      ? `(Caveat: ${this.fix(hdf.descriptions.caveat)})\r\n`
+      : '';
+    return `${caveat}${this.fix(hdf.wraps.desc)}`;
+  }
+
+  createTestResultDescription(hdf: HDFControl): string {
+    const controlStatus = `${hdf.status}:\r\n\r\n`;
+    const description =
+      hdf.segments?.map((result) => {
+        const statusAndTest = `${result.status.toUpperCase()} -- Test: ${
+          result.code_desc
+        }\r\n`;
+        const message = result.message
+          ? `Message: ${result.message}\r\n\r\n`
+          : '\r\n';
+        return `${statusAndTest}${message}`;
+      }).join('') ?? '';
+    return `${controlStatus}${description}`;
+  }
+
+  createTestResult(hdf: HDFControl): string {
+    return hdf.status === 'Passed' ? 'Satisfied' : 'Other Than Satisfied';
+  }
+
+  createImpact(hdf: HDFControl): string {
+    const controlSeverity =
+      hdf.severity === 'medium' ? 'moderate' : hdf.severity;
+    return this.fix(hdf.wraps.impact === 0 ? 'none' : controlSeverity);
+  }
+
+  getRow(control: ContextualizedControl, filename: string): CAATRow[] {
+    const hdf = control.hdf;
+    const allRows: CAATRow[] = _.compact(
+      hdf
+        .canonized_nist(FromHDFToCAATMapper.NistCanonizationConfig)
+        .map((formattedNistTag) => {
+          // I don't think the canonized_nist function would return something that wasn't a nist tag, but this check was there originally quite probably for a reason, but I've not been able to find a sample that triggers this case myself
+          if (!formattedNistTag) {
+            // early exiting forces us to use the compact function to get rid of empty values in the array
+            return;
+          }
+
+          const row: CAATRow = {};
+          row['Control Number'] = formattedNistTag;
+          row['Finding Title'] = `Test ${this.fix(hdf.wraps.id)} - ${this.fix(
+            hdf.wraps.title
+          )}`;
+          if (hdf.start_time) {
+            row['Date Identified'] = FromHDFToCAATMapper.formatDate(
+              new Date(hdf.start_time),
+              '/'
+            );
+          }
+          row['Finding ID'] = `${filename} - Test ${this.fix(hdf.wraps.id)}`;
```

Comment:
Do you need to add checks before referencing all of these nested fields, or is there a default that handles it? For example, what is hdf.wraps is null?

Reply:
https://github.com/mitre/heimdall2/blob/master/libs/inspecjs/src/compat_impl/compat_inspec_1_0.ts#L40

wraps can't be null.  if it were null, there would be no underlying data in the first place.  wraps is the name they chose for the attribute that references the original control more or less.  certain attributes like id are required for a control which is why you can access them without any checks.